Remotion/Web/Core/UI/Controls/SmartLabel.cs

```diff
@@ -184,12 +184,15 @@ protected virtual void LoadResources (IResourceManager resourceManager)
   private string? GetClientIDForTarget ()
   {
     Control? target = ControlHelper.FindControl(NamingContainer, ForControl);
-    if (target is ISmartControl && target is IFocusableControl)
+    if (target is ISmartControl smartTarget)
     {
-      if (((ISmartControl)target).UseLabel)
-        return ((IFocusableControl)target).FocusID;
+      if (!smartTarget.UseLabel)
+        return null;
+
+      if (target is IFocusableControl focusableControl)
+        return focusableControl.FocusID;
 
-      return null;
+      return target.ClientID;
```

Comment:
I updated the docs in https://re-motion.atlassian.net/browse/RM-8853 for the "best" implementation. I think it already covers most of the notes but I added a case for IFocusableControl without ISmartControl. Please verify the logic against the updated spec and check if you want to extract this into a separate bug. If you see something strange, please question :)